syntax = "proto2";
package event;

message SkuEventMessage {
    required string request_id = 1;
    required string event_label = 2;
    optional fixed64 created_at = 3;
    optional string created_by = 4;
    optional SkuMessage data = 5;
    optional string error = 6;
}

message SkuMessage {
    required string key = 1;
    required string code = 2;
    required string name = 3;
    required string specification = 4;
    required string uom = 5;
    optional string type = 6;
    optional string handling_list = 7;
    required Size size = 8;
    required Detail detail = 9;
    required string category = 10;
    required bool status = 11;
    optional fixed64 created_at = 12;
    optional string created_by = 13;
    optional fixed64 updated_at = 14;
    optional string updated_by = 15;
}

message Size{
    required SizePattern inner = 1;
    required SizePattern outer = 2;
    required SizePattern packages = 3;
}

message SizePattern{
    required uint32 length = 1;
    required uint32 width = 2;
    required uint32 height = 3;
    required uint32 weight = 4;
    required string size_category = 5;
}

message Detail{
    repeated Image images = 1;
}

message Image{
    required string url = 1;
}